ddtrace/tracer/baggage.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package tracer
+
+import (
+	"context"
+)
+
+// baggageKey is an unexported type used as a context key. It is used to store baggage in the context.
+// We use a struct{} so it won't conflict with keys from other packages.
+type baggageKey struct{}
+
+// baggageMap retrieves the map that holds baggage from the context.
+// Returns nil if no baggage map is stored in the context.
+func baggageMap(ctx context.Context) map[string]string {
+	val := ctx.Value(baggageKey{})
+	bm, ok := val.(map[string]string)
+	if !ok {
+		// bm is not a map[string]string or val was nil
+		return nil
+	}
+	return bm
```

Comment:
Sorry for confusion, but now that I'm seeing it split up like this, I'm realizing the old implementation is fine -- so long as every caller of `baggageMap` knows to check for nil. You can add this in the godoc comment, e.g. "Callers should check for a nil return value before invoking a method on the returned map". 
```suggestion
	bm, _ := ctx.Value(baggageKey{}).(map[string]string)
	return bm
```
Alternatively, you can have this `baggageMap` function return `(map[string]string, bool)` and have callers check the boolean for true (comma, ok idiom).